### PR TITLE
Adding missed OS check for GS support in Port Library for zLinux

### PR DESCRIPTION
**Pull request by ebadyano**
Update make files to be able to use auxv.c helpers for s390 Linux
builds.

Signed-off-by: Evgenia Badyanova <evgeniab@ca.ibm.com>

### CODE REVIEW

File: port/CMakeLists.txt [hunk 1]
```diff
@@ -213,6 +213,10 @@ if(OMR_HOST_ARCH STREQUAL ppc)
 	if(OMR_HOST_OS STREQUAL linux)
 		list(APPEND OBJECTS auxv.c)
 	endif()
+	elsif(OMR_ARCH_S390)
+	if(OMR_HOST_OS STREQUAL linux)
+		list(APPEND OBJECTS auxv.c)
+	endif()
```
[the remark this answers is not in the document]
Comment (ebadyano) [reply port/CMakeLists.txt]: i think my indentation is wrong, one of the endif() is for the nested if() inside elseif(). I'll add a missing white space.

**Comment by charliegracie** [comment]
@genie-omr build all

File: port/unix/auxv.c [hunk 2]
```diff
@@ -31,7 +31,9 @@
 #include <elf.h>
 #include <link.h>
 #include <pthread.h>
+#if defined(LINUXPPC)
```
Comment (charliegracie) [inline port/unix/auxv.c]: These changes seem to be enabling the compilation of this file on Linux 390.  How come we had to add new includes if we are compiling on Linux PPC?

Comment (ebadyano) [reply port/unix/auxv.c]: the include was there before and I assume compiles fine on power, but for some reason that header wasn't found when compiling on s390 Linux. It didn't seem to be necessary for query_auxv(AT_HWCAP). @ymanton Younes, do you know <asm/elf.h> is used in the lib on Power?

Comment (ymanton) [reply port/unix/auxv.c]: It's not necessary on ppc64le, probably not necessary on ppc64 or ppc32 either.

Comment (fjeremic) [reply port/unix/auxv.c]: > It's not necessary on ppc64le, probably not necessary on ppc64 or ppc32 either.

@ebadyano we should probably just remove the `#if defined` and the `#include` as well then?

Comment (ebadyano) [reply port/unix/auxv.c]: @ymanton Younce, is that okay with you, if we remove that include?

**Comment by fjeremic** [comment]
@genie-omr build all

File: port/unix/auxv.c (hunk 2 above)
Comment (ymanton) [reply port/unix/auxv.c]: Yes it's fine.

**Pull request merged**